map_test.go

```diff
@@ -88,6 +88,97 @@ func TestMap(t *testing.T) {
 	}
 }
 
+func TestBatchPerCPUHash(t *testing.T) {
+	if err := haveBatchAPI(); err != nil {
+		t.Skipf("batch api not available: %v", err)
+	}
+
+	m, err := NewMap(&MapSpec{
+		Type:       PerCPUHash,
```

Comment:
Yes, but there is a quirk.

`m.BatchLookup()` will now not return `ErrKeyNotExist` because it's not a "key", it's a value in an array. (This was a head scratcher for a while for me).

So that does mean we drop the testing of `batchLookupCmd()`
```
	sysErr = wrapMapError(sysErr)
	if sysErr != nil && !errors.Is(sysErr, unix.ENOENT) {
		return 0, sysErr
	}
```

Reply:
Yeah, I think I'll fix around this. I do realize the `Batch` API is one of those interesting dual return APIs that can be confusing, i.e. `count>0` and `error` is set.

There should be a case where `batch size == number of keys` and that should not return any error I think, it's only when we try to request one more that it fails.

Reply:
Take a look at the rewritten `TestMapBatch`: that now uses a `contents` map to assert that we get the correct values out. We don't test the order anymore, but for hashes that isn't defined and for arrays I'm not bothered enough to care. Net result is that the test is a lot more compact while testing both array and hash.